### ISSUE TITLE
$75 | “More” tab is not swapping position with the first element to the left of the tab

### ISSUE THREAD

**Issue by codejamtc** [topcoder-platform/navigation-component#101]
https://www.screencast.com/t/Gr2VNvQ8

1. Open the application https://test-community-app.topcoder-dev.com/challenges
2. Login as dan_developer (Landscape)
3. Tap Menu > Work > More > Community

**Actual:** “More” tab is not swapping position with the first element to the left of the tab

**Expected:** Selecting an option from the “More” tab should become visible and swap position with the first element to the left of the tab. This functionality can be observed on Google search.

**Reproducibility Rate:** 3/3

**Environment:** Samsung Galaxy Tab A (2017), 7 inch

**Browser:** Chrome 75.0.3770.80

**Comment by luizrrodrigues** [comment]
@yoution I see the swapping, but if click in level 3 menu will swap again to original position.

Please check:

[Screen Recording 2020-01-14 at 07.35.03.mov.zip](https://github.com/topcoder-platform/navigation-component/files/4058667/Screen.Recording.2020-01-14.at.07.35.03.mov.zip)

**Comment by yoution** [comment]
@luizrrodrigues      please review

**Comment by yoution** [comment]
https://github.com/topcoder-platform/navigation-component/pull/107

**Comment by luizrrodrigues** [comment]
@yoution If access link directly, like:
http://local.topcoder.com/community/qa

swapping not working, still showing Community -> Compete - More

**Comment by yoution** [comment]
@luizrrodrigues  https://github.com/topcoder-platform/navigation-component/pull/107
two commit , I add new commit,please review

**Comment by luizrrodrigues** [comment]
@yoution Looks like not working in Firefox, working well in another browsers.

**Comment by yoution** [comment]
@luizrrodrigues  https://monosnap.com/file/waNYOBDyc4onocBc66Y1UfDlBxyoHM
in my computer it work well

**Comment by luizrrodrigues** [comment]
@yoution I'll recheck.

**Comment by luizrrodrigues** [comment]
@yoution Looks good, I just redeploy from fresh repo and works well.

**Comment by luizrrodrigues** [comment]
@yoution please check: https://github.com/topcoder-platform/navigation-component/pull/107#issuecomment-575841552

**Comment by luizrrodrigues** [comment]
@yoution 

Please check: 

[Screen Recording 2020-01-21 at 22.07.26.mov.zip](https://github.com/topcoder-platform/navigation-component/files/4094759/Screen.Recording.2020-01-21.at.22.07.26.mov.zip)

Same related in #141 

This happen you go to same second level menu, ex.:
1) Select Tracks -> QA
2) Select Explore
3) Back to Tracks (see the issue)

**Comment by yoution** [comment]
@luizrrodrigues the green indicator not show correctly when switch tab？I will check

**Comment by luizrrodrigues** [comment]
Yes, Green indicator show in wrong position when follow the steps above.

**Comment by yoution** [comment]
@luizrrodrigues  please review
https://github.com/topcoder-platform/navigation-component/pull/128

**Comment by luizrrodrigues** [comment]
@yoution Please check video bellow, issue still exist:

[Screen Recording 2020-01-27 at 20.51.36.mov.zip](https://github.com/topcoder-platform/navigation-component/files/4119928/Screen.Recording.2020-01-27.at.20.51.36.mov.zip)

**Comment by yoution** [comment]
@luizrrodrigues please review again, thanks  https://github.com/topcoder-platform/navigation-component/pull/149

**Comment by luizrrodrigues** [comment]
@yoution Cool, looks good now.

**Comment by SathyaJayabal** [comment]
I am sorry but the issue in  #141 still happens. See video
<img width="798" alt="Screenshot 2020-01-28 at 6 42 49 PM" src="https://user-images.githubusercontent.com/58783823/73267446-18659d80-41ff-11ea-8c4f-b4d7ccd43081.png">
[more issue.mov.zip](https://github.com/topcoder-platform/navigation-component/files/4122551/more.issue.mov.zip)

**Comment by luizrrodrigues** [comment]
@yoution This happen just showing "More" button, because function will try replace menu in More with no existing out from More. 

So when screen in small and just More is showing, need ignore swipe function.

Make sense?

**Comment by SathyaJayabal** [comment]
@luizrrodrigues , Please make sure this issue is also fixed. 
On a tab size window, when you click on more->menu->sub menu, the selection indicator moves from more to Community
https://github.com/topcoder-platform/navigation-component/issues/62#issuecomment-567965305

[selection indicator.mov.zip](https://github.com/topcoder-platform/navigation-component/files/4122851/selection.indicator.mov.zip)

**Comment by yoution** [comment]
@luizrrodrigues  https://github.com/topcoder-platform/navigation-component/pull/150
have pushed, please review

**Comment by yoution** [comment]
@luizrrodrigues  I have pushed again, please review this one https://github.com/topcoder-platform/navigation-component/pull/151 . thanks

**Comment by macs054** [comment]
Verified in Beta
![image](https://user-images.githubusercontent.com/1199534/75420212-7dcdbb00-5972-11ea-8c4b-54adc47aa40b.png)
